### ISSUE TITLE
Linked Service (Azure Function) values hard coded in adf_publish ARM template

### ISSUE THREAD

**Issue by RedOystein** [Azure/Azure-DataFactory#93]
I have a ADF instance that is connected to a Azure DevOps Repo git repository, and is used as a part of a devops release pipeline. 

As a part of our ADF CI/CD pipeline we overwrite the generated `ARMTemplateForFactory.json` file that is produced when we click the Publish button in the ADF UI - which commits the ARM templates to `adf_publish`.

I have created a Azure Function Linked Service inside ADF, which is set up with a URL and a Function Key. The problem is that when the ADF instance is published, the Azure Function Linked Service inside the generated ARM templates does not create a parameter for the `functionAppUrl`, only for the `functionKey`. This makes it really hard for us to integrate ADF in our Azure DevOps CI/CD pipeline.

This is not the case with other Linked Services such as Oracle/SQL Server/Data Lake. For these Linked Services all the parameters that are needed to change the environment are inside the `ARMTemplateForFactory.json`.

Example of the ARM linked service resource created in the `adf_publish` branch.

```json
{
	"name": "[concat(parameters('factoryName'), '/AzureFunction1')]",
	"type": "Microsoft.DataFactory/factories/linkedServices",
	"apiVersion": "2018-06-01",
	"properties": {
		"annotations": [],
		"type": "AzureFunction",
		"typeProperties": {
			"functionAppUrl": "https://somerandom.azurewebsites.net",
			"functionKey": {
				"type": "SecureString",
				"value": "[parameters('AzureFunction1_functionKey')]"
			}
		}
	},
	"dependsOn": []
}
```

What it should create
```json
{
	"name": "[concat(parameters('factoryName'), '/AzureFunction1')]",
	"type": "Microsoft.DataFactory/factories/linkedServices",
	"apiVersion": "2018-06-01",
	"properties": {
		"annotations": [],
		"type": "AzureFunction",
		"typeProperties": {
			"functionAppUrl": "[parameters('AzureFunction1_functionAppUrl)]", <---
			"functionKey": {
				"type": "SecureString",
				"value": "[parameters('AzureFunction1_functionKey')]"
			}
		}
	},
	"dependsOn": []
}
```

**Comment by RedOystein** [comment]
I haven't read the documentation thorough enough, this can be fixed by adding a parameters definition file to the ADF repository.
https://docs.microsoft.com/en-us/azure/data-factory/continuous-integration-deployment#use-custom-parameters-with-the-resource-manager-template

**Comment by kirikintha** [comment]
I realize that this has been closed, but I would like to ask that this ticket be re-opened by the ADF administrators for consideration as a feature that can be added in future releases of Data Factory.

Not being able to set the location of the function app in a CI pipeline makes it very difficult to create a pipeline integration.

@RedOystein - I created an "arm-template-parameters-definition.json" file in the root of the repository and then added the following as per their documentation. This seemed to replicated the original information I had in the template parameters and add azure function url. 

https://gist.github.com/kirikintha/159c1101e41e920327e6d015ce5b7bdb
